v0.38.1 release (#2517)
grafana/k6: release notes/v0.38.1.md

k6 v0.38.1 is patch release containing a bugfix!

> ### Threshold expression containing reserved symbols would fail [#2512](https://github.com/grafana/k6/issues/2512)

Reviewer: ```suggestion
### Threshold sub-metric selectors containing reserved symbols would fail ([#2512](https://github.com/grafana/k6/issues/2512))
```

"threshold expressions" would probably more accurately refer to something like `p(99) < 500`

> There was a minor bug in threshold parsing which led to errors when users would use specific symbols such as `{`, `}` or `:` as part of their definitions. For instance, thresholds used in the context of [URL Grouping](https://k6.io/docs/using-k6/http-requests/#url-grouping) would have led to failures in `v0.38.0`. 

Reviewer: ```suggestion
There was a minor bug in threshold sub-metric selector parsing which led to errors when users would use specific symbols such as `{`, `}` or `:` as part of their definitions. For instance, thresholds used for sub-metrics with [URL Grouping](https://k6.io/docs/using-k6/http-requests/#url-grouping) like `http_req_duration{name:"http://example.com/${}"}` would have led to failures in `v0.38.0`. 

The error messages for invalid metric, sub-metric and threshold definitions were also improved.
```

> There was a minor bug in threshold parsing which led to errors when users would use specific symbols such as `{`, `}` or `:` as part of their definitions. For instance, thresholds used in the context of [URL Grouping](https://k6.io/docs/using-k6/http-requests/#url-grouping) would have led to failures in `v0.38.0`. 

Reviewer 2: ```suggestion
There was a bug in threshold parsing which led to errors when users would use specific symbols such as `{`, `}` or `:` as part of their definitions. For instance, thresholds used in the context of [URL Grouping](https://k6.io/docs/using-k6/http-requests/#url-grouping) would have led to failures in `v0.38.0`. 
```

> Special thanks to @efdknittlfrank, who reported, and helped us track down the issue.

Reviewer: ```suggestion
Special thanks to @efdknittlfrank, who reported and helped us track down the issue.
```
